### PR TITLE
chore: upgrade Go to 1.23.0

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ smoke: build
 smoke:
 	go test -v -tags='smoke' ./pkg/tests/smoke/...
 
-GOLANGCI_LINT_VERSION ?= v1.59.0
+GOLANGCI_LINT_VERSION ?= v1.60.1
```
Comment (thedadams) [inline Makefile:25]: Bump to support Go 1.23

File: pkg/loader/openapi.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func getOpenAPITools(t *openapi3.T, defaultHost, source, targetToolName string)
 		pathObj := pathMap[pathString]
 		// Handle path-level server override, if one exists
 		pathServer := defaultServer
-		if pathObj.Servers != nil && len(pathObj.Servers) > 0 {
+		if len(pathObj.Servers) > 0 {
```
Comment (thedadams) [inline pkg/loader/openapi.go:89]: New linter complaint.

File: pkg/openapi/getschema.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ func GetSchema(operationID, defaultHost string, t *openapi3.T) (string, Operatio
 	for path, pathItem := range t.Paths.Map() {
 		// Handle path-level server override, if one exists.
 		pathServer := defaultServer
-		if pathItem.Servers != nil && len(pathItem.Servers) > 0 {
+		if len(pathItem.Servers) > 0 {
```
Comment (thedadams) [inline pkg/openapi/getschema.go:86]: New linter complaint.

File: pkg/repos/runtimes/node/node.go [hunk 4]
```diff
@@ -136,7 +136,7 @@ func (r *Runtime) runNPM(ctx context.Context, tool types.Tool, toolSource, binDi
 		if tool.WorkingDir == "" {
 			return nil
 		}
-		if _, err := os.Stat(filepath.Join(tool.WorkingDir, packageJSON)); errors.Is(fs.ErrNotExist, err) {
+		if _, err := os.Stat(filepath.Join(tool.WorkingDir, packageJSON)); errors.Is(err, fs.ErrNotExist) {
```
Comment (thedadams) [inline pkg/repos/runtimes/node/node.go:139]: New linter complaint, which is actually a bug.

File: pkg/sdkserver/routes.go [hunk 5]
```diff
@@ -271,7 +271,7 @@ func (s *server) parse(w http.ResponseWriter, r *http.Request) {
 	} else {
 		content, loadErr := input.FromLocation(reqObject.File, reqObject.DisableCache)
 		if loadErr != nil {
-			logger.Errorf(loadErr.Error())
+			logger.Errorf("failed to load file: %v", loadErr)
```
Comment (thedadams) [inline pkg/sdkserver/routes.go:274]: New linter complaint.

**Review by njhale — APPROVED** [review]

**Review by g-linville — APPROVED** [review]

**Pull request merged**